lib/streamlit/config.py

```diff
@@ -982,6 +991,17 @@ def _check_conflicts():
             "another value, or remove it from your Streamlit config."
         )
 
+    # CSRF conflicts
+
+    if get_option("server.enableCSRFProtection"):
+        if not get_option("server.enableCORS") or get_option("global.useNode"):
+            LOGGER.warning(
+                "server.enableCSRFProtection is not compatible with server.enableCORS. "
+                "We will prioritize server.enableCSRFProtection over server.enableCORS "
+                "where CSRF has been enabled. If cross origin POST, PUT or "
+                "DELETE requests are required, please disable 'server.enableCSRFProtection'"
```

Comment:
>  If cross origin POST, PUT or DELETE requests are required, please disable 'server.enableCSRFProtection'

What does this mean? IIUC this is not a helpful message as the developer has no idea if what HTTP verbs we're using...